frontend/src/scenes/data-management/definition/definitionEditLogic.ts

```diff
@@ -71,7 +71,9 @@ export const definitionEditLogic = kea<definitionEditLogicType>([
                         const _eventProperty = definition as PropertyDefinition
                         definition = await api.propertyDefinitions.update({
                             propertyDefinitionId: _eventProperty.id,
-                            propertyDefinitionData: _eventProperty,
+                            propertyDefinitionData: values.hasTaxonomyFeatures
+                                ? _eventProperty
+                                : { property_type: _eventProperty.property_type },
```

Comment:
It was pretty confusing to me why this fixes anything. I know now, having found [this](https://github.com/PostHog/posthog/blob/0aa753edf3ab85a0932e6b1f11befcd69b097ba6/posthog/api/property_definition.py#L359) logic, but it would be great for the future if we had a comment here that ONLY `property_type` is editable for users without the premium feature, and including others fails for them.

Reply:
Right, I needed to look this up in the API first as well. Have added a comment as suggested.

Reply:
A potential alternative here is to handle this on the backend, where we remove keys that haven't changed, even if they are sent.

It errors out only when you send more than one key that has changed. Seems clearer to me.